x-pack/plugins/lens/public/editor_frame_service/editor_frame/frame_layout.tsx

```diff
@@ -18,67 +18,87 @@ export interface FrameLayoutProps {
   configPanel?: React.ReactNode;
   suggestionsPanel?: React.ReactNode;
   workspacePanel?: React.ReactNode;
+  bannerMessages?: React.ReactNode;
 }
 
 export function FrameLayout(props: FrameLayoutProps) {
   const isFullscreen = useLensSelector(selectIsFullscreenDatasource);
 
   return (
-    <EuiPage
-      className={classNames('lnsFrameLayout', {
-        'lnsFrameLayout-isFullscreen': isFullscreen,
-      })}
-    >
-      <EuiPageBody
-        restrictWidth={false}
-        className="lnsFrameLayout__pageContent"
-        aria-labelledby="lns_ChartTitle"
-      >
-        <section
-          className={classNames('lnsFrameLayout__sidebar lnsFrameLayout__sidebar--left', {})}
-          aria-labelledby="dataPanelId"
-        >
-          <EuiScreenReaderOnly>
-            <h2 id="dataPanelId">
-              {i18n.translate('xpack.lens.section.dataPanelLabel', {
-                defaultMessage: 'Data panel',
-              })}
-            </h2>
-          </EuiScreenReaderOnly>
-          {props.dataPanel}
-        </section>
-        <section
-          className={classNames('lnsFrameLayout__pageBody', {
-            'lnsFrameLayout__pageBody-isFullscreen': isFullscreen,
+    <EuiFlexGroup direction="column" responsive={false} gutterSize="none" alignItems="stretch">
+      {props.bannerMessages ? (
+        <EuiFlexItem grow={false}>
+          <section aria-labelledby="bannerMessagesId">
+            <EuiScreenReaderOnly>
+              <h2 id="bannerMessagesId">
+                {i18n.translate('xpack.lens.section.bannerMessages', {
+                  defaultMessage: 'Banner messages',
+                })}
+              </h2>
+            </EuiScreenReaderOnly>
+            {props.bannerMessages}
+          </section>
```

Comment:
I'm thinking that an `aside` element may be a more appropriate containing element for the warning callouts than a `section`. Thoughts on changing?